pkg/bootstrap/docker/openshift/metrics.go

```diff
@@ -0,0 +1,210 @@
+	addClusterReaderRole := policy.RoleModificationOptions{
+		RoleName:            "cluster-reader",
+		RoleBindingAccessor: clusterRoleBindingAccessor,
+		Users:               []string{"system:serviceaccount:openshift-infra:heapster"},
+	}
+	if err = addClusterReaderRole.AddRole(); err != nil {
+		return errors.NewError("cannot add cluster reader role to heapster service account").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	// Create metrics deployer secret
+	deployerSecret := &kapi.Secret{}
+	deployerSecret.Name = "metrics-deployer"
+	deployerSecret.Data = map[string][]byte{"nothing": []byte("/dev/null")}
+	if _, err = kubeClient.Secrets(infraNamespace).Create(deployerSecret); err != nil {
+		return errors.NewError("cannot create metrics deployer secret").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	// Create deployer Pod
+	imagePrefix := fmt.Sprintf("%s-", prefix)
+	deployerPod := metricsDeployerPod(hostName, imagePrefix, imageVersion)
+	if _, err = kubeClient.Pods(infraNamespace).Create(deployerPod); err != nil {
+		return errors.NewError("cannot create metrics deployer pod").WithCause(err).WithDetails(h.OriginLog())
+	}
+	return nil
+}
+
+func metricsDeployerPod(hostName, imagePrefix, imageVersion string) *kapi.Pod {
+	env := []kapi.EnvVar{
+		{
+			Name: "PROJECT",
+			ValueFrom: &kapi.EnvVarSource{
+				FieldRef: &kapi.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &kapi.EnvVarSource{
+				FieldRef: &kapi.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name:  "IMAGE_PREFIX",
+			Value: imagePrefix,
+		},
+		{
+			Name:  "IMAGE_VERSION",
+			Value: imageVersion,
+		},
+		{
+			Name:  "MASTER_URL",
+			Value: "https://kubernetes.default.svc:443",
+		},
+		{
+			Name:  "HAWKULAR_METRICS_HOSTNAME",
+			Value: hostName,
+		},
+		{
+			Name:  "MODE",
+			Value: "deploy",
+		},
+		{
+			Name:  "REDEPLOY",
+			Value: "false",
+		},
+		{
+			Name:  "IGNORE_PREFLIGHT",
+			Value: "false",
+		},
+		{
+			Name:  "USE_PERSISTENT_STORAGE",
+			Value: "false",
+		},
+		{
+			Name:  "CASSANDRA_NODES",
+			Value: "1",
+		},
+		{
+			Name:  "CASSANDRA_PV_SIZE",
+			Value: "10Gi",
+		},
+		{
+			Name:  "METRIC_DURATION",
+			Value: "7",
+		},
+		{
+			Name:  "HEAPSTER_NODE_ID",
+			Value: "nodename",
+		},
+		{
+			Name:  "METRIC_RESOLUTION",
+			Value: "10s",
+		},
+	}
+	pod := &kapi.Pod{
+		Spec: kapi.PodSpec{
+			DNSPolicy:          kapi.DNSClusterFirst,
+			RestartPolicy:      kapi.RestartPolicyNever,
+			ServiceAccountName: "metrics-deployer",
+			Volumes: []kapi.Volume{
+				{
+					Name: "empty",
+					VolumeSource: kapi.VolumeSource{
+						EmptyDir: &kapi.EmptyDirVolumeSource{},
+					},
+				},
+				{
+					Name: "secret",
+					VolumeSource: kapi.VolumeSource{
+						Secret: &kapi.SecretVolumeSource{
+							SecretName: "metrics-deployer",
+						},
+					},
+				},
+			},
+		},
+	}
+	pod.GenerateName = "metrics-deployer-"
```

Comment:
Hadn't given it much thought. The original template has a generate name. But in my case, I guess it doesn't make much sense. I'll switch it to a regular name.